### PR TITLE
Flutter 3.24 update, Add CarouselView examples to demo 3

### PR DESCRIPTION
**Pull request by QuncCccccc**
This PR is to include `CarouselView` to M3 demo app.

https://github.com/user-attachments/assets/7db76c06-aa7e-4461-9baa-51fd5c4e6b38

## Pre-launch Checklist

- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I read the [Contributors Guide].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.

### CODE REVIEW

**Review by domesticmouse — APPROVED** [review]
I'm assuming this will be held until 3.24 is stable?

/cc @ericwindmill

**Comment by domesticmouse** [comment]
Also, I'm curious why `master` channel is flailing. Also, the CI is failing to load. Wha?

**Comment by ericwindmill** [comment]
> I'm assuming this will be held until 3.24 is stable?
> 
> /cc @ericwindmill

Yes, I'm holding because this is the only update for 3.24

File: tool/flutter_clean_packages [hunk 1]
```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
```
Comment (ericwindmill) [inline tool/flutter_clean_packages:1]: I added this because I'm treating this PR as the Flutter 3.24 update branch

**Review by ericwindmill — APPROVED** [review]

**Pull request merged**